packages/camera/camera_web/lib/src/camera.dart

```diff
@@ -107,18 +124,36 @@ class Camera {
       }
     }
     videoElement.srcObject = null;
+    stream = null;
   }
 
   /// Captures a picture and returns the saved file in a JPEG format.
+  ///
+  /// Enables the device flash when taking a picture if the flash mode
+  /// is either [FlashMode.auto] or [FlashMode.always].
   Future<XFile> takePicture() async {
+    final shouldEnableTorchMode =
+        flashMode == FlashMode.auto || flashMode == FlashMode.always;
```

Comment:
I think this is going to be slightly odd behavior, but I'm not sure how this can be implemented, other than the [highly experimental fillLightMode constraint](https://developer.mozilla.org/en-US/docs/Web/API/PhotoCapabilities/fillLightMode).

Can we add a comment around here explaining why `takePicture` does slightly different things than `setFlashMode`? Somebody tracing this will see:

`setFlashMode set to auto (explicitly DOESN'T SET torchMode` -> `takePicture sets torchMode to true`

Reply:
Good point, will add a comment for that!